components/core/src/clp_s/search/Output.hpp

```diff
@@ -232,7 +231,42 @@ class Output : public FilterClass {
     bool evaluate_array_filter(
             FilterOperation op,
             DescriptorList const& unresolved_tokens,
-            std::string const& value,
+            std::string& value,
+            std::shared_ptr<Literal> const& operand
+    );
+
+    /**
+     * The implementation of evaluate_array_filter
+     * @param item
+     * @param op
+     * @param unresolved_tokens
+     * @param cur_idx
+     * @param value
+     * @param operand
+     * @return true if the expression evaluates to true, false otherwise
+     */
+    inline bool evaluate_array_filter(
+            ondemand::value& item,
+            FilterOperation op,
+            DescriptorList const& unresolved_tokens,
+            size_t cur_idx,
+            std::shared_ptr<Literal> const& operand
+    ) const;
+
+    /**
+     * The implementation of evaluate_array_filter
```

Comment:
Because we have three functions with the same name, maybe we can differentiate them a little bit in the description?

Reply:
Makes sense. I might also differentiate the names because it's kind of hard to read.